Sources/BranchSDK/NSError+Branch.m

```diff
@@ -115,4 +115,50 @@ + (BOOL)branchDNSBlockingError:(NSError *)error {
     return NO;
 }
 
++ (BOOL)branchVPNBlockingError:(NSError *)error {
+    if (error) {
+        NSError *underlyingError = error.userInfo[@"NSUnderlyingError"];
+        if (underlyingError) {
+
+            /**
+             `Domain=kCFErrorDomainCFNetwork Code=-1004` indicates that the connection failed because a connection can't be made to the host.
+             Reference: https://developer.apple.com/documentation/cfnetwork/cfnetworkerrors/kcfurlerrorcannotconnecttohost?language=objc
+             
+             `_kCFStreamErrorCodeKey=61` indicates that the connection was refused.
+             Reference: https://opensource.apple.com/source/xnu/xnu-792/bsd/sys/errno.h.auto.html
+             */
+            
+            BOOL isCouldntConnectErrorCode = [@(-1004) isEqual:@(underlyingError.code)];
+            BOOL isLocalHostErrorKey = [@(61) isEqual:error.userInfo[@"_kCFStreamErrorCodeKey"]];
+            
+            if ([self isConnectedToVPN] && isCouldntConnectErrorCode && isLocalHostErrorKey) {
+                return YES;
+            }
+        }
+    }
+    return NO;
+}
+
+/**
+ Helper method to which checks the device's internet proxy settings for common VPN protocol and interface substrings to determine if a VPN enabled.
+ https://developer.apple.com/documentation/cfnetwork/cfnetworkcopysystemproxysettings()
+ */
++ (BOOL)isConnectedToVPN {
+    NSDictionary *proxySettings = (__bridge NSDictionary *)(CFNetworkCopySystemProxySettings());
```

Comment:
Are there are any declarations or permissions that need to be granted to access this?

Reply:
Fortunately, no permissions are needed